contracts/scoring/dsf.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import { IScoringFunction } from "../interface/ScoringFunction.sol";
+import { HashingHub } from "../HashingHub.sol";
+import { Hub } from "../Hub.sol";
+
+contract DSF is IScoringFunction {
+    Hub public hub;
+
+    constructor(address hubAddress) {
+        require(hubAddress != address(0));
+        hub = Hub(hubAddress);
+    }
+
+    function calculateScore(uint256 distance, uint96 stake, uint32 a, uint32 b)
+        public
+        returns (uint32)
+    {
+        // VERIFY: uint256 -> uint32 casting
+        return uint32((a * stake) / (b * distance));
```

Comment:
Should be parametrized to essentially approach a polynomial form. The additional parameters needed, apart from "a" and "b" coefficients, are exponents for stake and distance, and constants in both the numerator and denominator  (all of which should also have their setters)

Reply:
So idea is to have quadratic polynomials division?

$(a * stake^2 + b * stake + c) \over (d * distance^2 + e * distance + f)$